pkg/ddevapp/config.go

```diff
@@ -130,10 +130,17 @@ func NewApp(appRoot string, includeOverrides bool) (*DdevApp, error) {
 	if app.Database.Type == "" {
 		app.Database = DatabaseDefault
 	}
+
 	if app.DefaultContainerTimeout == "" {
 		app.DefaultContainerTimeout = nodeps.DefaultDefaultContainerTimeout
 	}
 
+	// Migrate UploadDir to UploadDirs
+	if app.UploadDirDeprecated != "" {
```

Comment:
Manually tested this path and it failed with massive panic. Set `upload_dir: ../private` in a Drupal 9 project, then `ddev start` got
```
$ ddev start
runtime: goroutine stack exceeds 1000000000-byte limit
runtime: sp=0x140205c0350 stack=[0x140205c0000, 0x140405c0000]
fatal error: stack overflow

runtime stack:
runtime.throw({0x104b96709?, 0x1051f6420?})
	/opt/homebrew/Cellar/go/1.20.5/libexec/src/runtime/panic.go:1047 +0x40 fp=0x16b7dd3d0 sp=0x16b7dd3a0 pc=0x1046538c0
runtime.newstack()
	/opt/homebrew/Cellar/go/1.20.5/libexec/src/runtime/stack.go:1105 +0x468 fp=0x16b7dd580 sp=0x16b7dd3d0 pc=0x10466dae8
runtime.morestack()
	/opt/homebrew/Cellar/go/1.20.5/libexec/src/runtime/asm_arm64.s:316 +0x70 fp=0x16b7dd580 sp=0x16b7dd580 pc=0x104684130
```

Reply:
Yeah, recursive call while migrating, is fixed now